### PR TITLE
Add tls and metadata header docs for inference logger

### PR DESCRIPTION
**Pull request by sivanantha321**
## Proposed Changes 

- Documents https://github.com/kserve/kserve/pull/3863
- Documents https://github.com/kserve/kserve/pull/3635

### CODE REVIEW

**Comment by sivanantha321** [comment]
@ruivieira Can you help review the documentation ?

File: docs/modelserving/logger/logger.md [hunk 1]
```diff
@@ -154,6 +154,75 @@ kubectl logs $(kubectl get pod -l serving.knative.dev/service=message-dumper -o
...
+        "defaultUrl": "http://default-broker",
+        "caBundle": "example-com-ca",
+        "caCertFile": "service-ca.crt",
+        "tlsSkipVerify": false
+    }
+```
+
+### Considerations
+
+- If TLS is configured in ConfigMap, but the bundle is not available or the CA cert is not found, the logger will use plain-text HTTP with no TLS.
```
Comment (ruivieira) [inline docs/modelserving/logger/logger.md:217]: Would it be worth mentioning that this behaviour (no cert CM on namespace but TLS configured in `inferenceservice-config`) is in order to not break previous behaviour/expectation. Fine by me to not add it, though.

File: docs/modelserving/logger/logger.md [hunk 2]
```diff
@@ -154,6 +154,75 @@ kubectl logs $(kubectl get pod -l serving.knative.dev/service=message-dumper -o
...
+- If TLS is configured, but the logger's endpoint protocol scheme is of type http, the inference logger will use the plain-text HTTP with no TLS.
+
+### Constraints
+
+- The CA cert file should be in the same namespace as the InferenceService.
+- You can only specify one CA cert per namespace.
+- Since, the ConfigMap name and the CA cert file name are cluster wide configurations, it cannot be changed for each namespace.
```
Comment (ruivieira) [inline docs/modelserving/logger/logger.md:224]: Although it is written, I'm not sure if it's better to clarify that the CM values provide the rule for the CM *name* and CA *name* (ie where to find them), but CAs don't have to be the same across namespaces (only their name).
But, again, I'm fine with merging as it is.

**Review by ruivieira — APPROVED** [review]
LGTM, thanks @sivanantha321!

**Review by yuzisun — APPROVED** [review]
/lgtm
/approve

**Pull request merged**